PyART/models/seob.py

```diff
@@ -0,0 +1,139 @@
+            t         = t/(M * lal.MTSUN_SI)
+            for mode in hlm_seob.keys():
+                hlm_seob[mode] = hlm_seob[mode]/fac/nu
+
+        self._u   = t
+        hlm       = convert_hlm(hlm_seob)
+        self._hlm = hlm
+
+        self._hp, self._hc = wfu.compute_hphc(hlm, modes=list(hlm.keys()))
+        self._dyn = {'t'     : self.SEOB.model.dynamics[:,0],
+                     'r'     : self.SEOB.model.dynamics[:,1],
+                     'phi'   : self.SEOB.model.dynamics[:,2],
+                     'Pr'    : self.SEOB.model.dynamics[:,3],
+                     'Pphi'  : self.SEOB.model.dynamics[:,4],
+                     'E'     : self.SEOB.model.dynamics[:,5]*nu,
+                     'MOmega': self.SEOB.model.dynamics[:,6],
+                     }
+        return 0
+    
+    def check_pars(self):
+        """
+        Check and adjust pars dictionary
+        """
+        # If using geometric units, assuming total mass = 1
+        if self.pars["use_geometric_units"] == "yes":
+            if "mass1" not in self.pars:
+                self.pars["mass1"] = self.pars["q"]/(1. + self.pars["q"])
+            if "mass2" not in self.pars:
+                self.pars["mass2"] = 1./(1. + self.pars["q"])
+            # Convert initial frequency to physical units, save geometric in dict
+            self.pars["f22_start_geom"] = self.pars["f22_start"]
+            self.pars["f22_start"] = self.pars["f22_start"]/(self.pars["mass1"] + self.pars["mass2"])/lal.MTSUN_SI
+            # Rescale deltaT to avoid problems with ringdown frequencies; this assumes the default value, not great
+            self.pars["deltaT"] = self.pars["deltaT"]/100.
+        if "q" not in self.pars:
+            self.pars["q"] = self.pars["mass1"]/self.pars["mass2"]
+        
+        # If x, y spin components given, check that approximant is SEOBNRv5->P<-HM
+        if max([np.abs(self.pars["spin1x"]), np.abs(self.pars["spin1y"]), 
+                np.abs(self.pars["spin2x"]), np.abs(self.pars["spin2y"])]) > 1.e-10:
+            if self.pars["approximant"] != "SEOBNRv5PHM":
+                print("Switching to SEOBNRv5PHM for non-aligned spins.")
+                self.pars["approximant"] = "SEOBNRv5PHM"
+
+    def compute_energetics(self):
+        pars = self.pars
+        q    = pars['q']
+        q    = float(q)
+        nu   = q/(1.+q)**2
+        dyn  = self.dyn
+        E, j = dyn['E'], dyn['Pphi']
+        Eb   = (E-1)/nu
+        self.Eb = Eb
+        self.j  = j
+        return Eb, j
+    
+def convert_hlm(hlm):
+    """
+    Convert the hlm dictionary from SEOB to PyART notation
+    """
+    hlm_conv = {}
+    for key in hlm.keys():
+        A   = np.abs(hlm[key])
+        p   = -np.unwrap(np.angle(hlm[key]))
+        hlm_conv[key] = {'real': A*np.cos(p), 'imag': -1*A*np.sin(p),
+                                'A'   : A, 'p' : p,
+                                'h'   : A*np.exp(-1j*p)
```

Comment:
We have recently moved from `h` to a more generic `z` key for the complex strain. This is because we use the same key also for Psi4. Can you make the change here?

Reply:
Done